stack-macos.yaml

```diff
@@ -13,6 +13,7 @@ extra-deps:
   - llvm-hs-9.0.1
   - llvm-hs-pure-9.0.0
   - megaparsec-8.0.0
+  - prettyprinter-1.6.2
```

Comment:
Why the fixed version?

Reply:
I just copied `stack.yaml`. Is that okay?
https://github.com/google-research/dex-lang/blob/673b5fb74effa809c989be6bbfc50d5ebf9803e3/stack.yaml#L12-L16

Adding the `prettyprinter` dependency to `stack-macos.yaml` is important for consistent pretty-printing across macOS and Linux. Otherwise, `dex script show-tests.dx` would produce different table expression pretty-printing across the platforms.

Reply:
Oh yeah, that's great! I just didn't remember we pinned it there